src/javascripts/utility/debounce.js

```diff
@@ -0,0 +1,24 @@
+//Source: https://davidwalsh.name/javascript-debounce-function
+
+export function debounce(func, wait, immediate) {
+  let timeout;
+
+  return function executedFunction() {
+    let context = this;
+    let args = arguments;
+	    
+    let later = function() {
+      timeout = null;
+      if (!immediate) func.apply(context, args);
+    };
+
+    let callNow = immediate && !timeout;
+	
+    clearTimeout(timeout);
+
+    timeout = setTimeout(later, wait);
+	
+    if (callNow) func.apply(context, args);
+  };
+};
```

Comment:
nice. it does look like you dont use semicolons in the code you wrote yourself, so i'd remove them here to be consistent

Reply:
if this is working for you, awesome that something so lightweight works. it is from 2014 so it may miss edge cases. fwiw I more often see https://www.npmjs.com/package/lodash.debounce

Reply:
piggybacking on Ben's comment...looking into prettier for your code editor might make your code formatting life easier